### PR TITLE
example.go: mini refactor + improvements

### PR DESCRIPTION
**Pull request by odeke-em**
* Replaced pattern of fmt.Println + os.Exit with log.Fatal(f) i.e:
fmt.Println(string + errMsg + string)
fmt.Println(errMsg)
os.Exit(1)

with:
log.Fatalf(formatSpecifier, args...)

* Early return for unspecified flags to ease readability by putting
flag checks into a switch.

* Report all missing keys at once instead of one at a time
* Report what environment variables are needed in the error mesasges i.e:
+ Before:
```shell
$ go run example.go
Need to set the client ID
exit status 1
```
Right after setting it:
```shell
$ go run example.go
Need to set the client Key
exit status 1
```

+ After:
```shell
$ go run example.go
must set the client ID in environment variable: "HOUNDIFY_CLIENT_ID" or via commmandline flag: -id
must set the client key in environment variable: "HOUNDIFY_CLIENT_KEY" or via commandline flag: -key
exit status 1
```

This prevents the need for the user to have to open up the file to
figure out what environment variables they need to set. After this, we
can now compile binaries and ship them out for download; the users will be able
see what variables to set from the program output itself.

### CODE REVIEW

File: example.go [hunk 1]
```diff
@@ -26,29 +33,21 @@ func main() {
...
-		}
-	}
+	clientID := derefOrFetchFromEnv(clientIDFlag, envClientIDKey)
+	clientKey := derefOrFetchFromEnv(clientKeyFlag, envClientKeyKey)
 
+	var errsList []string
 	if clientID == "" {
-		fmt.Println("Need to set the client ID")
-		os.Exit(1)
+		errsList = append(errsList, fmt.Sprintf("must set the client ID in environment variable: %q", envClientIDKey))
```
Comment (bengadbois) [inline example.go]: Could this error message be modified to also mention command line flag option too, something like "must set the client ID in the environment variable %q or set via the command line flag -%s"

File: example.go [hunk 2]
```diff
@@ -26,29 +33,21 @@ func main() {
...
 	}
 	if clientKey == "" {
-		fmt.Println("Need to set the client key")
-		os.Exit(1)
+		errsList = append(errsList, fmt.Sprintf("must set the client key in environment variable: %q", envClientKeyKey))
```
Comment (bengadbois) [inline example.go]: Same as with the client ID flag comment

**Review by bengadbois — CHANGES_REQUESTED** [review]
Thanks for the PR! Just a couple of small comments

File: example.go (hunk 1 above)
Comment (odeke-em) [reply example.go]: Done, thanks.

File: example.go (hunk 2 above)
Comment (odeke-em) [reply example.go]: Done, thanks.

**Review by bengadbois — APPROVED** [review]

**Comment by bengadbois** [comment]
Thanks @odeke-em! 👍

**Comment by odeke-em** [comment]
Anytime! Thank you @bengadbois for the review and merge.

**Pull request merged**